src/server/engine_shard_set.cc

```diff
@@ -56,6 +106,41 @@ EngineShard::Stats& EngineShard::Stats::operator+=(const EngineShard::Stats& o)
   return *this;
 }
 
+// for now this does nothing
+uint64_t EngineShard::DoDefragIfRequired(uint64_t cursor) {
+  // Placeholder for now!
+  return cursor;
+}
+
+// the memory de-fragmenting task is as follow:
+//  1. Check if memory usage is high enough
+//  2. Check if diff between commited and used memory is high enough
+//  3. Check if we have memory changes (to ensure that we not running endlessly).
```

Comment:
I would like to omit item (3) for now.
also, I think the first condition should be - if we have the non-zero cursor  - continue running the task.

Reply:
For item 3 OK
The second issue, I though about this, the thing is, if the cursor is not 0 it would return in high priority, but then maybe we don't have this "issue" any more, so maybe we don't need to continue the scan at all..

Reply:
i am afraid stopping in the middle will create a bias: if you will run over 30% of the table and reduce the committed memory stat and stop because it just below the threshold then in a second it will cross the threshold and you will need to run the same 30% again and then it will again stop and then again and again. It's better to run till the end and evaluate the global condition only if you finished the scan for the whole table.